clouddriver-google-common/src/main/groovy/com/netflix/spinnaker/clouddriver/googlecommon/GoogleExecutor.groovy

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.clouddriver.googlecommon
+
+import com.google.api.client.googleapis.batch.BatchRequest
+import com.google.api.client.googleapis.services.AbstractGoogleClientRequest
+import com.google.api.client.http.HttpResponseException
+import com.netflix.spectator.api.Clock
+import com.netflix.spectator.api.Id
+import com.netflix.spectator.api.Registry
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+import java.util.concurrent.TimeUnit
+import javax.annotation.PostConstruct
+
+@Component
+class GoogleExecutor {
+  static Registry global_registry
+
+  static Registry getRegistry() {
+    return global_registry
+  }
+
+  @Autowired
+  Registry autowired_registry
+
+  @PostConstruct
+  public void bindGlobalRegistry() {
+    global_registry = autowired_registry
+  }
+
+
+  final static String TAG_BATCH_CONTEXT = "context"
```

Comment:
Can you drop a comment here and in `GoogleExecutorTraits` to explain the structure of this? Why isn't just the `Trait` class sufficient for this?